c/crossmemory.c

```diff
@@ -1192,204 +1200,245 @@ typedef struct PCHandlerParmList_tag {
 
 ZOWE_PRAGMA_PACK_RESET
 
+#define CMS_MAIN_PCSS_STACK_POOL_SIZE       1024
+#define CMS_MAIN_PCSS_RECOVERY_POOL_SIZE    8192
+
+#define CMS_STACK_SIZE                      65536
+#define CMS_STACK_SUBPOOL                   132
+
+#define CMS_MSG_QUEUE_MAIN_POOL_PSIZE       32768
+#define CMS_MSG_QUEUE_MAIN_POOL_SSIZE       4096
+#define CMS_MSG_QUEUE_FALLBACK_POOL_PSIZE   16384
+#define CMS_MSG_QUEUE_SUBPOOL               132
+
 #ifdef _LP64
 
 #pragma prolog(handlePCSS,\
-"        PUSH  USING                              \n\
-LSSPRLG  LARL  10,LSSPRLG                         \n\
-         USING LSSPRLG,10                         \n\
-         STORAGE OBTAIN,COND=YES,LENGTH=65536,CALLRKY=YES,SP=132,LOC=31,BNDRY=PAGE \n\
-         LTR   15,15                              \n\
-         BZ    LSSINIT                            \n\
-         LA    15,52                              \n\
-         EREGG 0,1                                \n\
-         PR                                       \n\
-LSSINIT  DS    0H                                 \n\
-         LGR   13,1                               \n\
-         USING PCHSTACK,13                        \n\
-         EREGG 1,1                                \n\
-         MVC   PCHSAEYE,=C'F1SA'                  \n\
-         MVC   PCHPLEYE,=C'RSPCHEYE'              \n\
-         STG   4,PCHPLLPL                         \n\
-         STG   1,PCHPLUPL                         \n\
-         LA    1,PCHPARM                          \n\
-         LA    13,PCHSA                           \n\
-         J     LSSRET                             \n\
-         LTORG                                    \n\
-         DROP  10                                 \n\
-LSSRET   DS    0H                                 \n\
-         DROP                                     \n\
+"        PUSH  USING                                                    \n\
+LSSPRLG  LARL  10,LSSPRLG          establish addressability using r10   \n\
+         USING LSSPRLG,10                                               \n\
+         LGR   5,4                 save latent parm as CPOOL uses r4    \n\
+         USING PCLPARM,5                                                \n\
+         LLGT  2,PCLPPRM1          load global area from latent parm    \n\
+         USING GLA,2                                                    \n\
+         SAM31                                                          \n\
```

Comment:
While not documented, CPOOL GET and FREE *support* AMODE(64) callers as long as SYSSTATE AMODE64=YES has been specified.